### PR TITLE
meta: notify #t-rustdoc Zulip stream on backport nominations

### PR DESCRIPTION
**Pull request by fmease**
In July '23, it was decided to handle rustdoc-specific backport nominations in t-rustdoc meetings going forward ([Zulip announcement](https://rust-lang.zulipchat.com/#narrow/stream/266220-t-rustdoc/topic/T-rustdoc.20backports/near/374828518)). However, t-rustdoc meetings are far too infrequent for them to address nominations on time (contrary to the weekly t-compiler meetings).

Hence GuillaumeGomez and I came to the conclusion that {beta,stable}-nominated rustdoc PRs should be dealt with on a case by case basis, e.g. on Zulip.

This PR attempts to partially automate this process. ~~Sadly, `triagebot` is not quite as flexible has I've hoped. Blocked on `triagebot` improvements (see the `FIXME`s in this PR).~~ (Fixed in rust-lang/triagebot#1791).

r? GuillaumeGomez

### CODE REVIEW

**Review by GuillaumeGomez — APPROVED** [review]
Approving the PR just to show my support as I think this is a very nice add! Please ping me once the other blockers have been solved.

**Comment by Urgau** [comment]
Maybe instead of hijacking the https://github.com/rust-lang/rust/labels/beta-nominated label rustdoc could use it's own label `beta-rustdoc-nominated` ?
This would have the advantage of being clearer about how should be responsible for the backport but also may avoid many unnecessary ping as many PRs have both `T-compiler` and `T-rustdoc` labels.

**Comment by fmease** [comment]
Hmm, I'm not super convinced that we need to introduce <kbd>rustdoc-{beta,stable}-{nominated,accepted}</kbd> (whatever the naming). I think it would needlessly complicate the backport process as T-release members would have to remember to not only look for <kbd>{beta,stable}-accepted</kbd> but also <kbd>rustdoc-{beta,stable}-accepted</kbd>. It's not the world but I'd say it can be avoided.

[There are only 7 <kbd>T-{rustdoc,compiler}</kbd><kbd>beta-accepted</kbd> PRs](https://github.com/rust-lang/rust/pulls?q=is%3Apr+label%3Abeta-accepted+label%3AT-rustdoc+label%3AT-compiler+is%3Aclosed) out of [77 <kbd>T-rustdoc</kbd><kbd>beta-accepted</kbd> PRs](https://github.com/rust-lang/rust/pulls?q=is%3Apr+label%3Abeta-accepted+label%3AT-rustdoc++is%3Aclosed+) which amounts to 9.1%, not that frequent. We can also update `triagebot` to support `exclude_labels = ["T-compiler"]` for `notify-zulip` hooks if we'd like to rule this out.

Although the idea of adding <kbd>I-rustdoc-nominated</kbd> is pretty good on its own. While we could teach rustdoc reviewers to add <kbd>I-rustdoc</kbd> to <kbd>{beta,stable}-nominated</kbd> PRs, I personally much prefer automation if possible.

**Comment by Urgau** [comment]
I'm not proposing to introduce `beta-rustdoc-accepted` just the nominated variants `{stable,beta}-rustdoc-nominated` as `{beta,stable}-accepted` is already IMO agnostic.

**Comment by Mark-Simulacrum** [comment]
T-release looks for *both* nominated and accepted labels, any changes here are somewhat painful (especially since backports are currently manual, and expressing the necessary combined query in GitHub would be... difficult).

You can read more about the process here - https://forge.rust-lang.org/release/backporting.html

So even just the nominated labels changing would be enough to lead to additional pain for the release team.

**Comment by fmease** [comment]
I've created a triagebot patch here: rust-lang/triagebot#1791. If it gets accepted it will unblock this PR.

**Review by GuillaumeGomez — APPROVED** [review]
Thanks for working on this (again)! :smile:

File: triagebot.toml
Comment (fmease) [inline triagebot.toml:420]: Contrary to rust-lang/rust#82689's self-proclaimed *hack*, rustdoc is now *occupying* the labels https://github.com/rust-lang/rust/labels/beta-nominated and https://github.com/rust-lang/rust/labels/stable-nominated. Swapping them around to make https://github.com/rust-lang/rust/labels/T-rustdoc the primary label would not allow us to register a hook for both https://github.com/rust-lang/rust/labels/beta-nominated **and** https://github.com/rust-lang/rust/labels/stable-nominated.

Comment (fmease) [reply triagebot.toml:420]: I could add a FIXME comment here noting this and suggesting patching `triagebot` to support “`[[notify-zulip."beta-nominated"]]`” etc. (i.e., allowing users to specify an array of actions instead of just one), do nothing or implement that patch right away, at your liking!

Comment (fmease) [reply triagebot.toml:423]: I've added FIXMEs.

**Pull request merged**
